Post settings excerpt

oguzkocer opened the pull request:
Fixes #5807. This is first of many post settings PRs. I wanted to start with something isolated, so picked the excerpt. The design is not fully implemented yet, this PR only handles the functionality of the excerpt as well as the excerpt dialog's design.

We might refactor a few things in the code to make them reusable, however I didn't want to over-engineer it. Those refactors will come when we need them in the following PRs, as well as a full design revamp of the post settings page. Feel free to ping me to ask if something's planned for further PRs to make sure I didn't miss anything. Thanks!

The excerpt is at the top of the page to make it easier to test, it'll move back to where it belongs in the further PRs.

To test:
* Go into the post settings page, tap on the excerpt and change it.
* After you change it, tap on it again, to make sure the dialog shows the updated text
* Remove the excerpt and make sure it's set to "Not Set"
* When the excerpt was removed, update the settings and make sure the excerpt on the remote is empty and not "Not Set"

tonyr59h commented on WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java:
```diff
@@ -351,7 +360,11 @@ private void setupSuggestionServiceAndAdapter() {
+        if (!TextUtils.isEmpty(mPost.getExcerpt())) {
+            mExcerptTextView.setText(mPost.getExcerpt());
+        } else {
+            mExcerptTextView.setText(getString(R.string.not_set));
```
👍

tonyr59h commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -278,14 +278,18 @@
     <string name="pages_fetching">Fetching pages…</string>
     <string name="toast_err_post_uploading">Unable to open post while it\'s uploading</string>
 
+    <!--post settings-->
```
nitpick; there's a `editor post settings` section on line 980.

tonyr59h commented:
Deleting the excerpt text after setting it doesn't show `Not Set`.

tonyr59h commented:
I should clarify that leaving the editor and returning it will correctly show `Not Set`.

oguzkocer commented:
> Deleting the excerpt text after setting it doesn't show `Not Set`.

Fixed in 83322e34bac450e45a2dfccdce3ccee805f5fc1a. I've also introduced `setPostExcerptToNotSet` method to avoid calling `mExcerptTextView.setText(R.string.not_set);` from multiple places.

Thanks for the review @tonyr59h, ready for another round!

tonyr59h commented:
Works now! :shipit:
